.github/workflows/ci.yml

```diff
@@ -0,0 +1,156 @@
+name: CI
+
+on: [pull_request, push]
+
+jobs:
+  build_and_test_windows:
+    name: Build and test
+    runs-on: ${{ matrix.os }}
+    env:
+      RUSTFLAGS: -Ctarget-feature=+crt-static
+      RUST_BACKTRACE: 1
+    strategy:
+      matrix:
+        os: [windows-latest]
+        rust: [stable]
+        target: [x86_64-pc-windows-msvc, i686-pc-windows-msvc, i686-pc-windows-gnu]
```

Comment:
We would want specific mingw version to be able to compile.
Take at look at my config here: 
https://github.com/lzutao/junction/blob/d21209cb8c34e0942cb4578937f0b92f76bbbfb5/.github/workflows/ci.yml#L13-L26